application/src/main/java/org/togetherjava/tjbot/features/basic/Starboard.java

```diff
@@ -0,0 +1,77 @@
+package org.togetherjava.tjbot.features.basic;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.Permission;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.entities.channel.concrete.TextChannel;
+import net.dv8tion.jda.api.entities.channel.middleman.GuildChannel;
+import net.dv8tion.jda.api.events.message.react.MessageReactionAddEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.config.StarboardConfig;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.features.EventReceiver;
+
+import java.util.Optional;
+
+import static org.togetherjava.tjbot.db.generated.tables.StarboardMessages.STARBOARD_MESSAGES;
+
+public class Starboard extends ListenerAdapter implements EventReceiver {
+
+    private static final Logger logger = LoggerFactory.getLogger(Starboard.class);
+    private final StarboardConfig config;
+    private final Database database;
+
+    public Starboard(Config config, Database database) {
+        this.config = config.getStarboard();
+        this.database = database;
+    }
+
+    @Override
+    public void onMessageReactionAdd(@NotNull MessageReactionAddEvent event) {
+        String emojiName = event.getEmoji().asCustom().getName();
+        Guild guild = event.getGuild();
+        if (ignoreMessage(emojiName, guild, event.getGuildChannel())) {
+            return;
+        }
+        Optional<TextChannel> starboardChannel = getStarboardChannel(guild);
+        if (starboardChannel.isEmpty()) {
+            logger.warn("There is no channel for the starboard in the guild with the name {}",
+                    config.getChannelName());
+            return;
+        }
+        if (database.read(context -> context.selectFrom(STARBOARD_MESSAGES)
+            .fetch("message_id")
+            .contains(event.getMessageIdLong()))) {
+            database.write(context -> context.newRecord(STARBOARD_MESSAGES)
+                .setMessageId(event.getMessageIdLong()));
+        }
+        event.retrieveMessage()
+            .flatMap(
+                    message -> starboardChannel.orElseThrow().sendMessageEmbeds(formEmbed(message)))
+            .queue();
+    }
+
+    private boolean ignoreMessage(String emojiName, Guild guild, GuildChannel channel) {
```

Comment:
bad method name. its missing a `shouldXXX`, otherwise it sounds as if this method, when called, will ignore the message